File: app/src/main/java/vergecurrency/vergewallet/viewmodel/WalletFragmentViewModel.java
package vergecurrency.vergewallet.viewmodel;

import androidx.lifecycle.ViewModel;
import vergecurrency.vergewallet.service.model.Currency;
import vergecurrency.vergewallet.service.model.PreferencesManager;
import vergecurrency.vergewallet.service.repository.PriceStatsDataReader;
import vergecurrency.vergewallet.wallet.WalletManager;

public class WalletFragmentViewModel extends ViewModel {
    private Long balance;
    //TODO check if necessary
    //private String currencyChange;

    public WalletFragmentViewModel() {
        balance = WalletManager.getBalance().getValue();
        //TODO check if necessary
        //currencyChange = PriceStatsDataReader.readPriceStatistics(getCurrencyCode()).get("XVG/USD");
    }

    public String getCurrencyCode() {
        return Currency.getCurrencyFromJson(PreferencesManager.getPreferredCurrency()).getCurrency();
    }


    public Long getBalance() {
        return balance;
    }

    public void setBalance(Long balance) {
        this.balance = balance;
    }
}

File: app/src/main/kotlin/vergecurrency/vergewallet/viewmodel/WalletFragmentViewModel.kt
package vergecurrency.vergewallet.viewmodel

import androidx.lifecycle.ViewModel
import vergecurrency.vergewallet.service.model.Currency
import vergecurrency.vergewallet.service.model.EncryptedPreferencesManager
import vergecurrency.vergewallet.service.repository.RatesClient
import vergecurrency.vergewallet.wallet.WalletManager

class WalletFragmentViewModel : ViewModel() {
    var balance: Long? = null
    //TODO check if necessary
    private val currencyChange: String?

    val currencyCode: ByteArray
        get() = Currency.getCurrencyFromJson(EncryptedPreferencesManager.preferredCurrency!!).currency!!

    init {
        balance = WalletManager.getBalance().value
        //TODO check if necessary
        currencyChange = RatesClient.infoBy(currencyCode)!!.price.toString()
    }
}